src/test/java/com/ibm/cloud/sdk/core/test/security/Cp4dAuthenticatorTest.java

```diff
@@ -281,14 +281,37 @@ public void testApiErrorBadRequest() throws Throwable {
 
     Request.Builder requestBuilder = new Request.Builder().url("https://test.com");
 
+    // Calling authenticate should result in an exception.
+    try {
+      authenticator.authenticate(requestBuilder);
+      fail("Expected authenticate() to result in exception!");
+    } catch (RuntimeException excp) {
```

Comment:
```suggestion
    } catch (ServiceResponseException excp) {
```
You could just catch the ServiceResponseException directly and call that a success, while anything else (the Throwable catch clause) would be an error.

Reply:
This should remain `RuntimeExcpetion` here, since the underlying error is IllegalStateException from the bad JSON. The ServiceResponse Exception changes have been made in both the `testApiErrorBadRequest()` locations where it is now surfaced.

Reply:
I was just testing you :)  HAHAHA